### DISCUSSION TITLE
How to support the tree structure?

### DISCUSSION THREAD

**Discussion by zhangking** [strapi/strapi#7093] [Legacy Topics]
Id | Name | Created_at | Updated_at |  
-- | -- | -- | -- | --
  | 1 | level1 | Tuesday, July 21st 2020 11:29 | Tuesday, July 21st 2020 11:29 |  
  | 2 | level2 | Tuesday, July 21st 2020 11:29 | Tuesday, July 21st 2020 11:29 |  
  | 3 | level3 | Tuesday, July 21st 2020 11:29 | Tuesday, July 21st 2020 11:31

I try  to make a tree  structure by using relation

level1 is related to level2 and level2 is related to level3

But I get the data through api

```
{
id: 1,
name: "level1",
tree:- {
id: 2,
name: "level2",
tree: 3,
created_at: "2020-07-21T03:29:34.325Z",
updated_at: "2020-07-21T03:29:52.397Z"
},
created_at: "2020-07-21T03:29:26.268Z",
updated_at: "2020-07-21T03:29:46.965Z"
}

```

It lose the data of level3

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Comment by derrickmehaffy** [comment]
Hi @zhangking you will need to customize the populate of your `find` service (docs here: https://strapi.io/documentation/v3.x/concepts/services.html#find) Please see the note there for the `populate` array. Let me know if you are still having issues.
